Modules/src/UserInputValidator.cc

```diff
@@ -44,6 +61,86 @@ namespace ChimeraTK {
     return rejected;
   }
 
+  void UserInputValidator::enableDeepValidation() {
+    // Find out accessors with return
+    for(auto& accessor : _module->getAccessorListRecursive()) {
+      if(accessor.getDirection() == VariableDirection{VariableDirection::feeding, true} &&
+          accessor.getModel().getTags().count(std::string(UserInputValidator::tagValidatedVariable)) > 0) {
+        _downstreamInvalidatingReturnChannels.emplace(accessor.getAppAccessorNoType().getId());
+      }
+    }
+
+    auto keepPvAccesWithReturnChannel = Model::EdgeFilter([](const Model::EdgeProperties& edge) -> bool {
+      return edge.type == Model::EdgeProperties::Type::pvAccess && edge.pvAccessWithReturnChannel;
+    });
+
+    // Filter that keeps our tagged variable nodes and application modules
+    auto validatedVariablesAndApplicationModulesFilter =
+        Model::VertexFilter([](const Model::VertexProperties& vertex) -> bool {
+          return vertex.visit([](auto props) -> bool {
+            if constexpr(Model::isVariable(props)) {
+              return props.tags.count(std::string(UserInputValidator::tagValidatedVariable)) > 0;
+            }
+
+            if constexpr(Model::isApplicationModule(props)) {
+              return true;
+            }
+
+            return false;
+          });
+        });
+
+    // Find longest path that is validated in our model, starting at this module
+    std::deque<Model::ApplicationModuleProxy> stack;
+    std::map<ApplicationModule*, int> distances;
+    distances[_module] = 0;
+
+    auto orderVisitor = [&](auto proxy) {
+      if constexpr(Model::isApplicationModule(proxy)) {
+        stack.push_front(proxy);
+        distances.try_emplace(&proxy.getApplicationModule(), std::numeric_limits<int>::min());
+      }
+    };
+
+    _module->getModel().visit(orderVisitor, Model::visitOrderPost, Model::depthFirstSearch,
+        keepPvAccesWithReturnChannel, validatedVariablesAndApplicationModulesFilter);
+
+    std::unordered_set<ApplicationModule*> downstreamModulesWithFeedback;
+
+    auto downstreamModuleCollector = [&](auto proxy) {
+      if constexpr(Model::isApplicationModule(proxy)) {
+        downstreamModulesWithFeedback.insert(&proxy.getApplicationModule());
+      }
+    };
+
+    auto connectingVariableVisitor = [&](auto proxy) {
+      if constexpr(Model::isVariable(proxy)) {
+        proxy.visit(downstreamModuleCollector, Model::adjacentOutSearch, validatedVariablesAndApplicationModulesFilter,
+            Model::keepApplicationModules);
```

Comment:
here you are passing two vertex filters to the same `visit()` call, which is not allowed. The `Model::keepApplicationModules` is redundant, as it is already part of the `validatedVariablesAndApplicationModulesFilter`. I am not 100% sure what this code tries to do, so it's hard to judge whether it is correct... Is it intentional that there is no `keepPvAccessWithReturnChannel`?

Reply:
That should have been keepPvAccessWIthReturnChannel, not validatedVariables... Fixed.